src/Microsoft.Data.SqlClient/src/Microsoft/Data/SqlClient/AlwaysEncryptedKeyConverter.cs

```diff
@@ -82,10 +35,16 @@ internal static RSAParameters RSAPublicKeyBlobToParams(byte[] keyBlob)
         {
             Debug.Assert(keyBlob.Length == RSAPublicKeyBlob.Size,
                 $"RSA public key blob was not the expected length. Actual: {keyBlob.Length}. Expected: {RSAPublicKeyBlob.Size}");
+
+            byte[] exponent = new byte[RSAPublicKeyBlob.ExponentSize];
+            byte[] modulus = new byte[RSAPublicKeyBlob.ModulusSize];
+            Buffer.BlockCopy(keyBlob, RSAPublicKeyBlob.ExponentOffset, exponent, RSAPublicKeyBlob.ExponentOffset, RSAPublicKeyBlob.ExponentSize);
+            Buffer.BlockCopy(keyBlob, RSAPublicKeyBlob.ModulusSize, modulus, RSAPublicKeyBlob.ModulusOffset, RSAPublicKeyBlob.ModulusSize);
```

Comment:
There is a `System.ArgumentException` related to these lines.

> Offset and length were out of bounds for the array or count is greater than the number of elements from index to the end of the source collection.

Probable fix:
> Buffer.BlockCopy(keyBlob, RSAPublicKeyBlob.ExponentOffset, exponent, 0, RSAPublicKeyBlob.ExponentSize);
> Buffer.BlockCopy(keyBlob, RSAPublicKeyBlob.ModulusOffset, modulus, 0, RSAPublicKeyBlob.ModulusSize);

Reply:
Good catch, fixed.